src/odemis/gui/conf/util.py

```diff
@@ -498,43 +498,52 @@ def format_choices(choices):
 
     return choices_formatted, choices_si_prefix
 
+def format_band_choices(comp, va, conf):
+    """
+    Formatter function for local axis VigilantAttributes
+    """
+
+    choices = va.choices
+
+    if not isinstance(choices, dict):
+        return va.choices
+
+    conf["unit"] = ""  # Disable unit display
+    return {k: to_readable_band(v) for k, v in va.choices.items()}
 
-def format_axis_choices(name, axis_def):
+
+def to_readable_band(v):
+    """
+    Convert a list of choices to readable bands for the GUI
+    """
+    if (isinstance(v, (tuple, list)) and len(v) > 1 and
+            all(isinstance(c, numbers.Real) for c in v)):
+        return fluo.to_readable_band(v)
+    else:
+        return v
+
+
+def format_axis_choices(name, choices, unit):
     """
     Transform the given choices for an axis into an user friendly display
 
     name (str): the name of the axis
-    axis_def (Axis): the axis definition
+    choices (dict): the choices.
 
     returns:
       choices_formatted (None or list of (value, str): axis value/user-friendly
          display name (including the unit). None if axis doesn't support choices.
     """
 
-    try:
-        choices = axis_def.choices
-    except AttributeError:
-        return None
-
     if not choices:
         return None
 
-    unit = axis_def.unit
-
     if isinstance(choices, dict):
```

Comment:
Please revert to the original version.

Reply:
Although now - we don't have axis_def anymore due to the changes of the function

Reply:
No, I mean you should revert the function signature and the internal code to be like it used to be (look it up in the git history, or on the github diff page). The only caller, which is in `create_axis_entry()` should also be reverted. 
The idea is that there is no need to change code that worked fine. This new version probably also works, but it's no necessary to mix it with the new feature you are adding.